app/assets/javascripts/reducers/feedback.js

```diff
@@ -15,12 +15,22 @@ export default function feedback(state = initialState, action) {
     }
     case POST_USER_FEEDBACK: {
       const newState = { ...state };
-      newState[action.assignmentId].custom.push({ message: action.feedback, messageId: action.messageId, userId: action.userId });
+      const assignmentId = action.assignmentId;
```

Comment:
The code for these actions is pretty complex and it's hard to reason about how it's supposed to work. Could you add a comment illustrating the typical structure of the `feedback` state? That would make the code easier to think about.

Reply:
@ragesoss I add comments to explain the structure of the feedback state. 

1. The original method uses push() to add an object to an array within an object in a state variable. It directly mutates the custom array inside the newState[action.assignmentId] object. I use the spread operator and object destructuring to create a new object with the updated custom array. It does not mutate the existing state but instead creates a new object with the updated custom array. 
2. Previous code uses splice(), which mutates the original array by removing elements and potentially adding new ones. I use filter() method to create a new array containing only the elements that pass the test implemented by the provided function. It does not modify the original array.

Hope this explains the revision clearly. Thank you.